File: CalenderView/app/src/main/java/com/non_name_hero/calenderview/data/CalendarData.java
package com.non_name_hero.calenderview.data;

import androidx.room.Ignore;

import java.util.Date;

public class CalendarData {
    public long scheduleId;
    public String scheduleTitle;
    public Date scheduleStartAtDatetime;
    public Date scheduleEndAtDatetime;
    public long groupId;
    public String groupTextColor;
    public int groupColorNumber;
    public int groupBackgroundColor;
    @Ignore
    public boolean isHoliday = false;
}

File: CalenderView/app/src/main/java/com/non_name_hero/calenderview/data/CalendarData.kt
package com.non_name_hero.calenderview.data

import androidx.room.Ignore
import java.util.*

class CalendarData {

    var scheduleId: Long = 0

    var scheduleTitle: String? = null

    var scheduleStartAtDatetime: Date? = null

    var scheduleEndAtDatetime: Date? = null

    var groupId: Long = 0

    var groupTextColor: String? = null

    var groupColorNumber = 0

    var groupBackgroundColor = 0


    @Ignore
    var isHoliday = false


    val textColor: Boolean
        get() = this.groupTextColor == "黒";
}